unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarShape.cs

```diff
@@ -126,6 +117,35 @@ public override IEnumerator ApplyChanges(BaseModel newModel)
             EnablePassport();
         }
 
+        private void UpdatePlayerStatus(AvatarModel model)
+        {
+            // Remove the player status if the userId changes
+            if (playerStatus != null && (playerStatus.id != model.id || playerStatus.name != model.name))
+                DataStore.i.player.otherPlayersStatus.Remove(playerStatus.id);
+
+            if (string.IsNullOrEmpty(model?.id))
+                return;
+
+            bool isNew = false;
+            if (playerStatus == null)
+            {
+                playerStatus = new PlayerStatus();
+                isNew = true;
+            }
+            playerStatus.id = model.id;
+            playerStatus.name = model.name;
+            playerStatus.isTalking = model.talking;
+            playerStatus.worldPosition = entity.gameObject.transform.position;
+            if (isNew)
+                DataStore.i.player.otherPlayersStatus.Add(playerStatus.id, playerStatus);
+        }
+
+        private void Update()
+        {
+            if (playerStatus != null)
+                playerStatus.worldPosition = entity.gameObject.transform.position;
```

Comment:
Couldn't we do this position update in the `OnEntityTransformChanged()` instead of do it in each frame?

Reply:
The entity transform position is not where the Avatar is rendered, we interpolate the target position to make it smoothly. We actually have to do it each frame or the AvatarName will be lagging behind. A nice "side-effect" is that the minimap is now smoother as well (since it can use this data).